frontend/src/services/userService.ts

```diff
@@ -1,66 +1,61 @@
 import Constants from 'expo-constants';
-import { setItem, STORAGE_KEYS, getItem } from './storage';
-import { apiFetch } from './apiClient';
-import { v4 as uuidv4 } from 'uuid';
+import { getItem, setItem, removeItem } from './storage';
+import { STORAGE_KEYS } from './storage';
 
-const IDIOMS_BACKEND_URL = Constants.expoConfig?.extra?.API_URL;
-
-const API_ROUTES = {
-  USERS: 'users/',
-} as const;
-
-const handleApiError = async (response: Response) => {
-  const text = await response.text();
-  throw new Error(`Server error ${response.status}: ${text.slice(0, 200)}`);
-};
-
-const generateUUID = (): string => uuidv4();
-
-const getOrCreateInstallationId = async (): Promise<string> => {
-  const existing = await getItem(STORAGE_KEYS.INSTALLATION_ID);
-  if (existing) return existing;
-
-  const id = generateUUID();
-  await setItem(STORAGE_KEYS.INSTALLATION_ID, id);
-  return id;
-};
-
-export interface RegisterUserResponse {
-  id: string;
-  installation_id: string;
-  api_key: string;
-  created_at: string;
-  updated_at: string;
-}
+const BACKEND_URL = Constants.expoConfig?.extra?.API_URL;
 
 export const registerOrGetApiKey = async (): Promise<string | null> => {
-  const saved = await getItem(STORAGE_KEYS.API_KEY);
-  if (saved) return saved;
+  const existingApiKey = await getItem(STORAGE_KEYS.API_KEY);
+  if (existingApiKey) {
+    return existingApiKey;
+  }
 
-  if (!IDIOMS_BACKEND_URL) {
-    console.warn(
-      'No backend URL configured (Constants.expoConfig.extra.API_URL)',
-    );
+  try {
+    const response = await fetch(`${BACKEND_URL}/users/register`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'ngrok-skip-browser-warning': 'idioms',
+      },
+      body: JSON.stringify({}),
+    });
+
+    if (!response.ok) return null;
+
+    const data = await response.json();
+    const apiKey = data.user.api_key;
+
+    await setItem(STORAGE_KEYS.API_KEY, apiKey);
+    return apiKey;
+  } catch {
     return null;
   }
+};
 
-  const installation_id = await getOrCreateInstallationId();
-
-  const path = `${API_ROUTES.USERS}register`;
-  const response = await apiFetch(path, {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' },
-    body: JSON.stringify({ installation_id }),
-  });
-
-  if (!response.ok) return handleApiError(response);
-
-  const data = (await response.json()) as RegisterUserResponse;
-
-  if (data && typeof data.api_key === 'string' && data.api_key) {
-    await setItem(STORAGE_KEYS.API_KEY, data.api_key);
-    return data.api_key;
+export const getCurrentUser = async () => {
+  const apiKey = await getItem(STORAGE_KEYS.API_KEY);
+  if (!apiKey) return null;
+
+  try {
+    const response = await fetch(`${BACKEND_URL}/users/me`, {
```

Comment:
I don't think this endpoint exists yet, does it? Are we using it in the front-end anywhere?